### PR TITLE
Feature/fga/load room members by chunk

### LINKED ISSUE #6394: LoadRoomMemberTask should insert by Chunk
In big rooms, while calling LoadRoomMemberTask, it can takes a lot of time to insert in db all the RoomMember events, ending up blocking all other db transactions. 
We should split the insertions by smaller chunks so we can  process the other transactions between each chunk.

### PR DESCRIPTION
**Pull request by ganfra**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

When handling the LoadRoomMembers response, we now make chunks of 500 events to avoid blocking the db too long in large rooms.

## Motivation and context

Fix #6394

## Tested devices

- [x] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/LoadRoomMembersTask.kt [hunk 1]
```diff
@@ -105,32 +106,37 @@ internal class DefaultLoadRoomMembersTask @Inject constructor(
...
+                        root = eventEntity
+                    }
+                    roomMemberEventHandler.handle(realm, roomId, roomMemberEvent, false)
+                }
+            }
+        }
         monarchy.awaitTransaction { realm ->
-            // We ignore all the already known members
             val roomEntity = RoomEntity.where(realm, roomId).findFirst()
                     ?: realm.createObject(roomId)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/LoadRoomMembersTask.kt:136]: Changing to `RoomMembersLoadStatusType.LOADED` is not atomic anymore with the room member insertion, but at worth case the SDK will load the room members one more time so this is fine.

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt [hunk 2]
```diff
@@ -83,23 +82,19 @@ internal class RoomMemberEventHandler @Inject constructor(
...
+            existingRoomMemberSummary.displayName = roomMember.displayName
+            existingRoomMemberSummary.avatarUrl = roomMember.avatarUrl
+            existingRoomMemberSummary.membership = roomMember.membership
+        } else {
+            val roomMemberEntity = RoomMemberEntityFactory.create(
+                    roomId,
+                    userId,
+                    roomMember
+            )
+            realm.insert(roomMemberEntity)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/membership/RoomMemberEventHandler.kt:96]: Thanks for the refacto. Read it twice, looks fine.

**Review by bmarty — APPROVED** [review]
LGTM, thanks!

**Review by Claire1817 — APPROVED** [review]
LGTM :)

**Pull request merged**